### PR TITLE
Added extension search text length to telemetry

### PR DESCRIPTION
**Pull request by prashantvc**
This PR adds the length of search text to the telemetry

### CODE REVIEW

**Comment by isidorn** [comment]
This looks ok to me, but I leave the final review to @sandy081 as he would know best if this catches all the cases where we have the `searchText` .

@prashantvc did you try this out, if it catches the data you need in all cases. To build VS Code out of source just follow this doc https://github.com/microsoft/vscode/wiki/How-to-Contribute
And then you can check the telemetry output log to see if the query length is being captured as you intended.

**Comment by prashantvc** [comment]
> And then you can check the telemetry output log to see if the query length is being captured as you intended.

Yes, I tried it locally on my machine! It seems to capture the data that I need

**Review by sandy081 — CHANGES_REQUESTED** [review]

File: src/vs/platform/extensionManagement/common/extensionGalleryService.ts [hunk 1]
```diff
@@ -329,6 +331,10 @@ class Query {
 		return new Query({ ...this.state, source });
 	}
 
+	withSearchTextLength(length: number): Query {
+		return new Query({ ...this.state, searchTextLength: length });
+	}
+
```
[the remark this answers is not in the document]
Comment (prashantvc) [reply src/vs/platform/extensionManagement/common/extensionGalleryService.ts:337]: Thanks for review fixing them now

**Comment by isidorn** [comment]
@prashantvc let us know when this is good for review again. Thank you

**Comment by prashantvc** [comment]
> let us know when this is good for review again. Thank you

It's ready now

**Review by sandy081 — APPROVED** [review]
LGTM - Thanks

**Review by isidorn — APPROVED** [review]

**Comment by isidorn** [comment]
Looks good, thanks a lot @prashantvc 
Let's merge this in.

**Pull request merged**